### DISCUSSION TITLE
Equipable projectiles

### DISCUSSION THREAD

**Discussion by ghost** [AscensionGameDev/Intersect-Engine#1480] [Ideas]
**Is your feature request related to a problem? Please describe.**
It would be interesting if the Arrow or any projectile can be equipable, and can be used in the Shield Slot

**Describe the solution you'd like**
To create a type of item, Projectile, the engine automatically allows you to equip it even using a bow, however, you will have to create one for staves too, for the player not to equip arrows. OR a type of BOW item, the arrow can only equip when a bow is equipped

**Comment by pandinocoder** [comment]
Restored issue.

**Comment by Cheshire92** [comment]
This feels awfully game specific to me?
Thoughts?

**Comment by Arufonsu** [comment]
> Thoughts?

I have seen this behaviour before, on games like Ragnarok Online, ie. when you equip specific arrows / shurikens / bullets / spears and they appear in the equipment inventory, but they are still stored in the item inventory. In the other hand theres Tibia, ie. there's a slot in the equipment inventory that's only for projectiles such as arrows / crossbow's bolts / throwing stars / etc. Unlike RO, when you wear a projectile, it frees up the inventory slot where the projectiles were stacked at.

![tibia](https://user-images.githubusercontent.com/17498701/107855257-13599900-6dd6-11eb-8d72-fec6cf7adbd4.png)
_`Example of the MMORPG "Tibia" - Ammo slot in the equipment window. Drag and Drop from Item Inventory (bags in the case of Tibia), basically  : [from ItemInventorySlot > to EquipmentInventorySlot]`_

I have played in a lot of classic MMORPGs (like the two mentioned in this comment) that follow this structure, specially when it comes to games where you have different kind of ammo for the same type of weapons (lets say, fire arrows / ice arrows or whatever arrow for a single bow type of weapon), then its way more clean and organized to have them equipped into the equipment inventory (thus, you know which ammo you are specifically using at the moment), than having all of them unorganized in the item inventory.

![RO](https://user-images.githubusercontent.com/17498701/107855786-3f2a4e00-6dd9-11eb-9244-2452ec4dbb34.jpg)
_`RO Example: Specific kind of arrow (Steel Arrows), being displayed as "Equipped" in RO's Equipment Inventory. Pretty much a Must if you have multiple kind of ammos for bows like this.`_

Just a final though on this comment, because its been implemented by other MMORPGs in the past, it means that its not a "specific" feature at some point anymore, thus, we shouldn't limit this engine and discard the idea.

Having this and more organization-friendly features as **optional** for those devs who wish to have their player's inventory sorted out differently would be nice, they could always opt out of these features anyway.

**Comment by blinkuz** [comment]
It is a basic functionality that would be good to add to the engine, in my specific case I have some items of knives and shurikens among a lot of tools that can be equipped by players that allow them to launch them, but consuming said item from the inventory in each launch.

**Comment by PyroTech03** [comment]
This would allow us to have projectiles that have different damages and effects and another way we can balance ranged combat. Poison/fire/ice arrows, all able to apply damage and a spell effect. Being able to be shot by the same bow. Instead of multiple bow types.
